server/controllers/resolve.js

```diff
@@ -9,13 +9,14 @@ const getAllExistingItemsForContract = require('../lib/get-all-existing-items-fo
 const syndicate = require('../lib/syndicate-content');
 
 module.exports = exports = async (req, res, next) => {
+	const log = new Logger({req, res, source: 'controllers/resolve'});
 
 	const { body } = req;
 
 	const { locals: { contract, flags } } = res;
 
 	if (!Array.isArray(body)) {
-		log.warn({
+		log.warn(`Expected \`req.body\` to be [object Array] and got \`${Object.prototype.toString.call(body)}\` instead`, {
```

Comment:
Preferably, this should be a shorter phrase without spaces or punctuation (using underscores like `RESOLVE_INVALID_REQUEST_BODY` or camelCase).

The detailed message is already contained in the data object. Also, this would make writing queries and grouping errors together easier.